packages/autocomplete-js/src/highlight.ts

```diff
@@ -39,17 +22,11 @@ type HighlightItemParams<TItem> = {
 export function highlightHit<TItem extends Hit<{}>>({
   hit,
   attribute,
-  highlightPreTag = '<mark>',
-  highlightPostTag = '</mark>',
-  ignoreEscape,
+  tagName = 'mark',
+  createElement = preactCreateElement,
 }: HighlightItemParams<TItem>) {
-  return concatParts(
-    parseAlgoliaHitHighlight<TItem>({
-      hit,
-      attribute,
-      ignoreEscape,
-    }),
-    { highlightPreTag, highlightPostTag }
+  return parseAlgoliaHitHighlight<TItem>({ hit, attribute }).map((x) =>
+    x.isHighlighted ? createElement(tagName, { children: x.value }) : x.value
```

Comment:
not needed yet today, but you might want class names on the tag, wdyt?

Reply:
Yep! I was thinking about spreading props in there or accepting a `tagProps` param if people need access to this.